tests/MiddlewareTest.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+namespace Spatie\Permission\Test;
+
+use Auth;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Spatie\Permission\Middleware\RoleMiddleware;
+use Spatie\Permission\Middleware\PermissionMiddleware;
+
+class MiddlewareTest extends TestCase
+{
+    /**
+     * @test
+     * @expectedException Symfony\Component\HttpKernel\Exception\HttpException
+     */
+    public function it_can_not_access_with_middleware_role_if_is_guest()
+    {
+        $request = new Request();
+        $middleware = new RoleMiddleware($this->app);
+
+        $result = $middleware->handle($request, function ($request) {
```

Comment:
That block

```
$middleware->handle($request, function ($request) {
             return (new Response())->setContent('<html></html>');
         }, 'testRole');
```

is something that is used a lot.

Wrap it into a protected helper method called `testMiddleware` or something better.

```php
function testMiddleware($middleware, $parameter) {
   $middleware->handle($request, function () {
        return (new Response())->setContent('<html></html>');
    }, $parameter)
}
```